modules/fluid_properties/src/userobjects/TwoPhaseFluidPropertiesIndependent.C

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TwoPhaseFluidPropertiesIndependent.h"
+
+registerMooseObject("FluidPropertiesApp", TwoPhaseFluidPropertiesIndependent);
+
+template <>
+InputParameters
+validParams<TwoPhaseFluidPropertiesIndependent>()
+{
+  InputParameters params = validParams<TwoPhaseFluidProperties>();
+
+  params.makeParamRequired<UserObjectName>("fp_liquid");
+  params.makeParamRequired<UserObjectName>("fp_vapor");
+
+  return params;
+}
+
+TwoPhaseFluidPropertiesIndependent::TwoPhaseFluidPropertiesIndependent(
+    const InputParameters & parameters)
+  : TwoPhaseFluidProperties(parameters)
+{
+  _fp_liquid = &_fe_problem.getUserObject<SinglePhaseFluidProperties>(_liquid_name);
+  _fp_vapor = &_fe_problem.getUserObject<SinglePhaseFluidProperties>(_vapor_name);
```

Comment:
This should be possible in the c-tor initializer list.

`GeneralUserObject`s now inherit from `UserObjectInterface`, so it should be possible to use just `getUserObject` and `hasUserObject` (which is used in `TwoPhaseFluidProperties`).

Reply:
Regarding the constructor, I don't think I can do this - they have already been initialized in the base class. I tried it, and I got the following error:
```
member initializer '_fp_liquid' does not name a non-static data member or base class
```